### ISSUE TITLE
Add a new category to release.yaml for dependencies

### ISSUE THREAD

**Issue by cris-oddball** [department-of-veterans-affairs/notification-api#1359]
## User Story - Business Need
Our [release.yaml](https://github.com/department-of-veterans-affairs/notification-api/blob/master/.github/release.yaml) file contains categories that become the release notes, based on the labels applied to PRs. The "dependencies" label does not have a category, so it ends up in "Other". This story is to create a header in the release notes based on the things in release.yaml such that dependencies have their own cool section in the release notes.

- [x] Ticket is understood, and QA has been contacted (if the ticket has a QA label).

### User Story(ies)

**As a**    <br> VANotify client or engineer
**I want**  <br> to see in the release notes a section for Dependencies
**So that**  I can quickly see what libraries were upgraded in that release.

## Engineering Checklist

- [x] The .github/release.yaml has a new section for dependencies
```
    - title: Dependency Upgrades <some icon here>
      labels:
        - dependencies
```
- [x] Pick a cool fun icon for the dependencies

## Acceptance Criteria

- When code is released to Prod and there were dependencies upgraded in that release, the Release Notes clearly show the Dependency Upgrades section, with a cool icon, and the dependency PR/s are listed.

**Comment by mjones-oddball** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/VA-Notify-620d21369d810a00146ed9c8/issues/department-of-veterans-affairs/notification-api/1359?planning-poker) with Zenhub @cris-oddball @justaskdavidb2 @k-macmillan @kalbfled @ldraney @nikolai-efimov

**Comment by cris-oddball** [comment]
Second PR approved, merged and deployed to perf on tag `v1.6.8-1376-dependency-category-2`

A fake pre-release was made from this tag and the release notes configured to go back far enough to grab a dependencies update:
![Screen Shot 2023-08-04 at 3.09.46 PM.jpg](https://images.zenhubusercontent.com/630696b479f5cca51e546872/6e40f184-302e-40e2-acef-7d1522c20777)

Closing ticket as complete!
